neqo-http3/src/connection_client.rs

```diff
@@ -2364,4 +2430,676 @@ mod tests {
+        );
+
+        assert_eq!(
+            client.state(),
+            Http3State::Closing(CloseError::Application(265))
+        );
+        assert_eq!(*server.state(), State::Connected);
+    }
+
+    #[test]
+    fn zero_rtt_new_server_setting_omit_header_list_size() {
+        // Connect, send a server settings with MaxHeaderListSize=10000
+        // and get a token
+        let (mut client, mut server) = connect();
+        let token = exchange_token(&mut client, &mut server.conn);
+
+        // Reconnect
+        let (mut client, mut server) = zero_rtt_server_setting(token, true);
+        // Send a new server settings without MaxHeaderListSize
+        server_side_make_control_qpack_streams_and_settings(
+            &mut client,
+            &mut server,
+            &[
+                (HSettingType::MaxTableSize, 100),
+                (HSettingType::BlockedStreams, 100),
+            ],
+        );
+
+        assert_eq!(client.state(), Http3State::Connected);
+        assert_eq!(*server.state(), State::Connected);
+    }
+
+    #[test]
+    fn zero_rtt_new_server_setting_max_table_size_bigger() {
+        // Connect, send a server settings with MaxTableSize=100
+        // and get a token
+        let (mut client, mut server) = connect();
+        let token = exchange_token(&mut client, &mut server.conn);
+
+        // Reconnect
+        let (mut client, mut server) = zero_rtt_server_setting(token, true);
+        // Send a new server settings MaxTableSize=200
+        server_side_make_control_qpack_streams_and_settings(
+            &mut client,
+            &mut server,
+            &[
+                (HSettingType::MaxTableSize, 200),
+                (HSettingType::BlockedStreams, 100),
+                (HSettingType::MaxHeaderListSize, 10000),
+            ],
+        );
+
+        assert_eq!(client.state(), Http3State::Connected);
+        assert_eq!(*server.state(), State::Connected);
+    }
+
+    #[test]
+    fn zero_rtt_new_server_setting_max_table_size_smaller() {
+        // Connect, send a server settings with MaxTableSize=100
+        // and get a token
+        let (mut client, mut server) = connect();
+        let token = exchange_token(&mut client, &mut server.conn);
+
+        // Reconnect
+        let (mut client, mut server) = zero_rtt_server_setting(token, true);
+        // Send a new server settings MaxTableSize=50
+        server_side_make_control_qpack_streams_and_settings(
+            &mut client,
+            &mut server,
+            &[
+                (HSettingType::MaxTableSize, 50),
+                (HSettingType::BlockedStreams, 100),
+                (HSettingType::MaxHeaderListSize, 10000),
+            ],
+        );
+
+        assert_eq!(
+            client.state(),
+            Http3State::Closing(CloseError::Application(265))
+        );
+        assert_eq!(*server.state(), State::Connected);
+    }
+
+    #[test]
+    fn zero_rtt_new_server_setting_blocked_streams_bigger() {
```

Comment:
There is a lot of repetition in these tests.  They should each reduce to "server starts with config X, resumes with config Y, result is Z".  These repeat a lot of boilerplate.